### DISCUSSION TITLE
Any way to add chunk options, especially "output: asis", to R inline code?

### DISCUSSION THREAD

**Discussion by setseed13** [quarto-dev/quarto-cli#4333] [Q&A] [answered]
Hi all!

I wanted to make my life easier by wrapping `<span style="color: blue">word</span>` into an R function.
```{r}
#| echo: false
colorize <- function(string, color) {
  string_colorized <- paste("<span style=\"color: ", color,"\">", string,"</span>",sep = "")
  cat(string_colorized)
}
```

The above works fine via:

```{r}
#| echo: false
#| output: asis
colorize("word", "blue")
```

However, this creates a new paragraph for a single, mid-sentence word, which lead me to throwing the entire paragraph into a slightly more complicated function.

Since this doesn't feel right, I wanted to ask whether there is any way to generate raw HTML via inline code?
Even when setting `output: asis` in the YAML, `r colorize("some text", "blue")` renders into nothing.

I also tried generating R inline code via sprintf, but, unsurprisingly, this also didn't work:

```{r}
#| echo: false
colorize_2 <- function(string, color) {
    string_colorized <- paste("`r sprintf(\"<span style='color: %2$s'>%1$s</span>\", \"", string, "\", \"", color, "\")`", sep = "")
    cat(string_colorized)
}
```

I'm thankful for any assistance!

**Comment by cderv** [comment, marked as answer]
Don't use `cat()` . 

This should work

````markdown
---
format: html
---

## Test

```{r}
colorize <- function(string, color) {
  string_colorized <- paste("<span style=\"color: ", color,"\">", string,"</span>",sep = "")
  string_colorized
}
```

This text is `r colorize("red", "red")` and then black again.

````
![image](https://user-images.githubusercontent.com/6791940/218500244-c6046975-16bc-4497-8e70-cff22ebfa0e1.png)


Such colorize function with R is shown as example in the R Markdown Cookbook by the way
https://bookdown.org/yihui/rmarkdown-cookbook/font-color.html#using-an-r-function-to-write-raw-html-or-latex-code

It will work with Quarto too

**Comment by setseed13** [comment]
I thought I lost my sanity, because I was pretty sure I saw a working example somewhere in the documentation...I was scouring the one for *Quarto* quite extensively...

And since the same holds for the LUA filters, you not only made my day, but my entire month!
Thank you so much!
